src/data/fin.lean

```diff
@@ -353,6 +353,10 @@ instance add_comm_monoid (n : ℕ) : add_comm_monoid (fin (n + 1)) :=
 
 end monoid
 
+/-- In the trivial ring, 0 = 1. We state it as `1 = 0` in the simplifying direction,
+because there are more lemmas where `0` is cancelling (in multiplication). -/
```

Comment:
Why do you need this in addition to `one_eq_zero_of_subsingleton`?

I think this stuff is just large enough a can of worms to put in it's own PR, so that all the obviously fine lemma can be merged

Reply:
This is necessary here because the ring (comm_ring) instance for `fin` is added in a much later file. I will separate the lemmas out.